Applications/WebPortal/WebApp/library/source/pages/account_page/http_account_model.ts

```diff
@@ -10,13 +10,18 @@ import { LocalAccountModel } from './local_account_model';
 export class HttpAccountModel extends AccountModel {
 
   /** Constructs an HttpAccountModel.
+   * @param authUserAccount - The authenticated user's account.
```

Comment:
What is an authenticated user's account?

Reply:
That represents the user that's currently logged in.

Reply:
What is it needed for?

Reply:
To compare the account roles for the user that's logged in and the account they're linking to. It's needed for evaluating isReadonly for the password change fields.

Reply:
Okay I see where you're coming from and it sounds reasonable. Look to move this logic out of the model and into the controller. The way I like to think of it is that the model manages data access and the controller decides what to do with that data. When I see something like whether something is read only, that to me sounds like a decision that needs to be made rather than some kind of data access.